miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -392,30 +391,26 @@ export.validate_procedure_metadata
             dup exec.get_procedure_metadata
             # => [storage_offset, storage_size, index, num_storage_slots, num_account_procedures]
 
-            # assert that storage offset is not 0
-            dup push.0 neq assert.err=ERR_FAUCET_INVALID_STORAGE_OFFSET
+            # Procedures that do not access storage are defined with (offset, size) = (0, 0).
+            # But we want to fail on tuples defined with a zero size but non-zero offset, since that is a logic error.
+            # We want to panic when we see this condition, so we use assertz rather than assert.
+            dup.1 eq.0 dup.1 eq.0 not and assertz.err=ERR_ACCOUNT_INVALID_STORAGE_OFFSET_FOR_SIZE
+            # => [storage_offset, storage_size, index, num_storage_slots, num_account_procedures]
```

Comment:
Similar to the above comment, this is equivalent to `(size == 0 & offset != 0) == 0`, right? If so, I'd include this expression (or something similar) in the comment.